### PR TITLE
Utils tests

### PR DESCRIPTION
**Pull request by poliorcetics**
- [x] Refactor the `utils` module to make it partially testable.
- [x] Add tests for the `utils` module.

### CODE REVIEW

**Comment by poliorcetics** [comment]
I want to add tests with line numbers next but the current tests are done. This is not ready to be merged though

**Comment by arijit79** [comment]
First of all, thanks again for your help. When you're done, mark the PR as done. Since there are some major changes, I would go for a manual check before merging

File: src/utils.rs [hunk 1]
```diff
@@ -1,64 +1,112 @@
...
+        MoveTo(0, rows as u16),
+        Attribute::Reverse,
+        Attribute::Reset,
+    );
+
+    drop(out);
+    let _ = stdout.flush();
+}
+
+/// Writes the given lines to the given `out`put.
```
Comment (arijit79) [inline src/utils.rs:39]: Why are you writing about these variables in doc comments. These will not be caught by rustdoc as the function is private. I would suggest converting them to simple comments

File: src/utils.rs [hunk 2]
```diff
@@ -1,64 +1,112 @@
...
+///
+/// - `rows` is the number of lines the `out`put can display at once.
```
Comment (arijit79) [inline src/utils.rs:41]:
```suggestion
// - `rows` is the number of lines the terminal can display at once.
```

Rows is the number of rows the terminal can display at once and not the output

File: src/utils.rs (hunk 1 above)
Comment (poliorcetics) [reply src/utils.rs]: I use `cargo doc --document-private-items` to navigate the crate structure easily and have a general idea of where things are. Having documentation on private items is very nice then.

File: src/utils.rs (hunk 2 above)
Comment (poliorcetics) [reply src/utils.rs]: The `out` parameter for this function is not necessarily a terminal, for example it's a `String` in the tests (that's what makes the function testable).

**Comment by poliorcetics** [comment]
For I left the unhandled-`Result`s as-is. I will address them in a follow-up when (if) this merged.

**Comment by arijit79** [comment]
Thanks for the PR. I have merged your changes

**Pull request merged**